flair/trainers/plugins/functional/checkpoints.py

```diff
@@ -30,8 +30,6 @@ def after_training_epoch(self, epoch, **kw):
             )
             model_name = "model_epoch_" + str(epoch) + ".pt"
             self.model.save(self.base_path / model_name, checkpoint=self.save_optimizer_state)
-            if torch.distributed.is_initialized():
-                torch.distributed.barrier()  # Prevent any process from loading a model until writing is complete
```

Comment:
It lead to a deadlock that froze the app (explanation in the upstream PR). It would be good muscle memory to have a barrier whenever you save, but isn't strictly needed since there isn't another process that's going to read the saved checkpoint file.